typedb/connection/transaction.py

```diff
@@ -50,10 +50,7 @@ def __init__(self, session: "_TypeDBSessionImpl", transaction_type: TransactionT
         self._logic_manager = _LogicManager(self)
 
         try:
-            # Other TypeDBClient implementations reuse a single gRPC Channel, but the Python client stalls
-            # when opening several transactions in parallel from one Channel.
-            stub = session.client().connection_factory().newTypeDBStub(session.client().connection_factory().newChannel(session.address()))
-            self._bidirectional_stream = BidirectionalStream(stub, session.transmitter())
+            self._bidirectional_stream = BidirectionalStream(session.client().stub(), session.transmitter())
```

Comment:
Reusing an existing stub rather than creating a new one.

You wrote a comment above the line saying that doing so may cause the client to stall, but the test seem to pass without stalling. Therefore it seems that my change is working, right @alexjpwalker ?

Reply:
Hmm, actually, I don't think this bug is covered in BDD.

It's covered in an integration test, which I stupidly seem to have forgotten to merge into master! See: https://github.com/alexjpwalker/typedb-client-python/blob/stalling-fix/tests/integration/test_concurrent.py

It opens 10000 transactions in parallel, closing each one -at some point- after opening.

Let's reintroduce the test and try it out locally?